### PR TITLE
Rename `VariantZeroAssigner` to `VariantDefaultInitializer`

### PR DESCRIPTION
**Pull request by aaronfranke**
Discussed on RocketChat. `VariantZeroAssigner`'s name indicates that it should assign zero, but it actually assigns the default value. For some types, such as Basis, Transforms, Color, Quat, the default value is non-zero. The new name `VariantDefaultInitializer` was suggested by @akien-mga.

definit

### CODE REVIEW

**Comment by TokageItLab** [comment]
I am wondering if this PR will make `Variant::zero()` obsolete. To begin with, there are few parts that use `Variant::zero()`, and it is only needed for blending `ValueTrack`s in `AnimationTree`. So, perhaps the right way to go would be to obsolete `Variant::zero()` and implement `AnimationTree::set_variant_to_zero()` as a local function of `AnimationTree`.

File: core/variant/variant_internal.h [hunk 1]
```diff
@@ -1201,176 +1201,176 @@ struct VariantInitializer<Object *> {
...
 
 template <class T>
-struct VariantZeroAssigner {
+struct VariantDefaultInitializer {
 };
 
 template <>
-struct VariantZeroAssigner<bool> {
+struct VariantDefaultInitializer<bool> {
 	static _FORCE_INLINE_ void zero(Variant *v) { *VariantInternal::get_bool(v) = false; }
```
Comment (akien-mga) [inline core/variant/variant_internal.h:1209]: If this is approved, then `zero` should also be renamed everywhere. But let's see what @reduz thinks first.

**Comment by akien-mga** [comment]
Discussed in PR review meeting, @reduz (and maybe @vnen) need to check a bit deeper why this is needed and why it doesn't reset all types of Variants (e.g. String, Callable).

**Comment by Iniquitatis** [comment]
Don't know how much of a compatibility breakage it will cause (if any), but can the `zero` method be renamed accordingly? IIRC, `default` is a reserved keyword, so what about `default_value`, for example?
As of now, it looks a bit weird that `DefaultInitializer` implenents the `zero` method, which doesn't zero out a value in all cases.

**Comment by vnen** [comment]
The change makes sense to me. Since it's an internal change it won't break compatibility. I do agree that the `zero()` methods should also be renamed.

**Comment by AThousandShips** [comment]
I'd just call it `init`, it is an `Initializer` and the other `Initializer`s call it that

**Review by akien-mga — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**